server/lobbyconnection.py

```diff
@@ -505,14 +515,69 @@ async def check_policy_conformity(self, player_id, uid_hash, session, ignore_res
 
         return response.get("result", "") == "honest"
 
+    async def command_auth(self, message):
+        token = message["token"]
+        unique_id = message["unique_id"]
+        player_id = await self.oauth_service.get_player_id_from_token(token)
+
+        new_irc_password = hexlify(os.urandom(16)).decode()
+        await self.send({
+            "command": "irc_password",
+            "password": new_irc_password
+        })
```

Comment:
Crazy thought. @Brutus5000 can we make the IRC server use hydra as well?

Reply:
You mean something like send a token instead of a username/password?
I looked this up for unrealircd and anope but no matches. Since OAuth is very web-centric I guess OAuth is considered work of the devil for IRC power users :wink: